app/src/main/java/com/example/frolic/EventDetailsActivity.java

```diff
@@ -0,0 +1,209 @@
+package com.example.frolic;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
+
+import java.util.ArrayList;
+
+/**
+ * Activity for displaying detailed information about a specific event.
+ */
+public class EventDetailsActivity extends AppCompatActivity {
+
+    private TextView tvEventName, tvEventDescription, tvEventDate, tvOrganizer;
+    private Button btnBack;
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private String eventId;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_event_details);
+
+        // Initialize views
+        tvEventName = findViewById(R.id.tvEventName);
+        tvEventDescription = findViewById(R.id.tvEventDescription);
+        tvEventDate = findViewById(R.id.tvEventDate);
+        tvOrganizer = findViewById(R.id.tvOrganizer);
+        btnBack = findViewById(R.id.btnBack);
+
+        // Set up the back button to finish the activity and go back
+        btnBack.setOnClickListener(v -> finish());
+
+        // Get eventId from Intent extras
+        eventId = getIntent().getStringExtra("eventId");
+
+        // Load event details from Firestore
+        if (eventId != null) {
+            loadEventDetails(eventId);
+        }
+    }
+
+    /**
+     * Loads event details from Firestore based on the provided eventId.
+     *
+     * @param eventId The ID of the event to load.
+     */
+    private void loadEventDetails(String eventId) {
+        DocumentReference eventRef = db.collection("events").document(eventId);
+        eventRef.get().addOnSuccessListener(documentSnapshot -> {
+            if (documentSnapshot.exists()) {
+                Event event = documentSnapshot.toObject(Event.class);
+                if (event != null) {
+                    tvEventName.setText(event.getEventName());
+                    tvEventDescription.setText(event.getEventDesc());
+                    tvEventDate.setText(event.getEventDate().toString());
+                    tvOrganizer.setText("Organized by: " + event.getOrganizerId());
```

Comment:
It might look nicer to display the organizer's name here instead of ID.

Reply:
Okay just changed that